receiver/k8sclusterreceiver/watcher.go

```diff
@@ -83,33 +78,108 @@ func newResourceWatcher(
 func (rw *resourceWatcher) prepareSharedInformerFactory() {
 	factory := informers.NewSharedInformerFactoryWithOptions(rw.client, 0)
 
-	// Add shared informers for each resource type that has to be watched.
-	rw.setupInformers(&corev1.Pod{}, factory.Core().V1().Pods().Informer())
-	rw.setupInformers(&corev1.Node{}, factory.Core().V1().Nodes().Informer())
-	rw.setupInformers(&corev1.Namespace{}, factory.Core().V1().Namespaces().Informer())
-	rw.setupInformers(&corev1.ReplicationController{},
-		factory.Core().V1().ReplicationControllers().Informer(),
-	)
-	rw.setupInformers(&corev1.ResourceQuota{}, factory.Core().V1().ResourceQuotas().Informer())
-	rw.setupInformers(&corev1.Service{}, factory.Core().V1().Services().Informer())
-	rw.setupInformers(&appsv1.DaemonSet{}, factory.Apps().V1().DaemonSets().Informer())
-	rw.setupInformers(&appsv1.Deployment{}, factory.Apps().V1().Deployments().Informer())
-	rw.setupInformers(&appsv1.ReplicaSet{}, factory.Apps().V1().ReplicaSets().Informer())
-	rw.setupInformers(&appsv1.StatefulSet{}, factory.Apps().V1().StatefulSets().Informer())
-	rw.setupInformers(&batchv1.Job{}, factory.Batch().V1().Jobs().Informer())
-	rw.setupInformers(&batchv1.CronJob{}, factory.Batch().V1().CronJobs().Informer())
-	rw.setupInformers(&autoscalingv2beta2.HorizontalPodAutoscaler{},
-		factory.Autoscaling().V2beta2().HorizontalPodAutoscalers().Informer(),
-	)
+	// Map of supported group version kinds by name of a kind.
+	// If none of the group versions are supported by k8s server for a specific kind,
+	// informer for that kind won't be set and a warning message is thrown.
+	// This map should be kept in sync with what can be provided by the supported k8s server versions.
+	// List of group version kinds must be ordered starting from the most recent versions.
+	// Informer will be configured to use the latest version supported by k8s API.
+	supportedKinds := map[string][]schema.GroupVersionKind{
```

Comment:
Would this map ever need to be used somewhere else?  Is there benefit to defining it in such a way that something else can reference it?

Reply:
No, this map is only needed in this function